engine/class_modules/sc_death_knight.cpp

```diff
@@ -6139,6 +6137,24 @@ struct scourge_strike_base_t : public death_knight_melee_attack_t
   int n_targets() const override
   { return p() -> in_death_and_decay() ? as<int>( p() -> spec.scourge_strike -> effectN( 4 ).base_value() ) : 0; }
 
+  std::vector<player_t*>& target_list() const override // smart targeting to targets with wounds when cleaving SS 
+  {
+    std::vector<player_t*>& current_targets = death_knight_melee_attack_t::target_list();
+    if (current_targets.size() < 2 || !p() -> in_death_and_decay()) {
+      return current_targets;
+    }
+    
+    // first target, the action target, needs to be left in place
+    std::sort( current_targets.begin() + 1, current_targets.end(),
+      [this]( player_t* a, player_t* b) {
+        int a_stacks = p()->get_target_data( a )->debuff.festering_wound->check();
```

Comment:
You can use the simpler death_knight_action_t::td( target ). It's a shortcut for p() -> get_target_data( t ).
``int a_stacks = td( a ) -> ...``